### PR TITLE
logging: posthog events added

### PR DESCRIPTION
**Pull request by FBalint**

### CODE REVIEW

File: src/components/[guild]/AddRewardButton/useCreateTokenReward.ts [hunk 1]
```diff
@@ -63,10 +70,18 @@ const useCreateReqBasedTokenReward = ({
...
+      captureEvent(
+        "Failed to add token reward (multiple roles selected)",
+        postHogOptions
+      )
       showErrorToast("Dynamic token rewards can be added to only one role at most.")
       return
     }
     if (data.requirements.length > 1) {
+      captureEvent(
+        "Failed to add token reward (multiple requirements set when adding to existing role)",
```
Comment (BrickheadJohnny) [inline src/components/[guild]/AddRewardButton/useCreateTokenReward.ts:82]: In order to keep these event names short, we could just call them "Failed to add token reward" & pass the reason (e.g. "multiple requirements set when adding to existing role) in an event property ;)

File: src/components/[guild]/AddRewardButton/useCreateTokenReward.ts [hunk 2]
```diff
@@ -82,6 +97,11 @@ const useCreateReqBasedTokenReward = ({
          * needed in the reward's rolePlatform's dynamicData field.
          */
 
+        captureEvent("createTokenReward(AddToExistingRole) Requirement created", {
```
Comment (BrickheadJohnny) [inline src/components/[guild]/AddRewardButton/useCreateTokenReward.ts:100]: Not sure if we need this one

File: src/components/[guild]/AddRewardButton/useCreateTokenReward.ts [hunk 3]
```diff
@@ -117,9 +137,23 @@ const useCreateReqBasedTokenReward = ({
           { revalidate: false }
         )
 
-        onAddRewardSubmit(rewardSubmitData).then(() => triggerMembershipUpdate())
+        onAddRewardSubmit(rewardSubmitData).then(() => {
```
Comment (BrickheadJohnny) [inline src/components/[guild]/AddRewardButton/useCreateTokenReward.ts:140]: Let's put it in the hook's `onSuccess` param instead

File: src/components/[guild]/AddRewardButton/useCreateTokenReward.ts [hunk 4]
```diff
@@ -138,6 +172,20 @@ const useCreateReqBasedTokenReward = ({
       rolePlatforms: [],
       logic: "AND",
       guildId,
+    }).catch((error) => {
```
Comment (BrickheadJohnny) [inline src/components/[guild]/AddRewardButton/useCreateTokenReward.ts:175]: Consider using `useCreateRole`'s `onError` instead

File: src/components/[guild]/AddRewardButton/useCreateTokenReward.ts [hunk 5]
```diff
@@ -165,7 +213,21 @@ const useCreateReqBasedTokenReward = ({
       saveAs,
       createdRole.id
     )
-    await onAddRewardSubmit(rewardSubmitData)
+    const createdReward = await onAddRewardSubmit(rewardSubmitData).catch(
+      (error) => {
```
Comment (BrickheadJohnny) [inline src/components/[guild]/AddRewardButton/useCreateTokenReward.ts:217]: I think we could also use the hook's `onError` prop here

File: src/platforms/Token/EditTokenModal.tsx [hunk 6]
```diff
@@ -106,6 +107,12 @@ const EditTokenModal = ({
   const { data: requirements } = useRequirements(role.id)
   const snapshotRequirement = requirements?.find((req) => !!req?.data?.snapshot)
 
+  const { captureEvent } = usePostHogContext()
```
Comment (BrickheadJohnny) [inline src/platforms/Token/EditTokenModal.tsx:110]: If we won't allow users to add this reward in the edit role drawer, we don't need events here ;)

File: src/platforms/Token/hooks/useCollectToken.tsx [hunk 7]
```diff
@@ -52,7 +59,17 @@ const useCollectToken = (
         method: "POST",
         body: {},
       },
-    ])
+    ]).catch((error) => {
```
Comment (BrickheadJohnny) [inline src/platforms/Token/hooks/useCollectToken.tsx:62]: We could use the hook's `onError` here too

File: src/platforms/Token/EditTokenModal.tsx (hunk 6 above)
Comment (FBalint) [reply src/platforms/Token/EditTokenModal.tsx:110]: But they can be edited through the access hub card, are you sure?

File: src/platforms/Token/hooks/useCollectToken.tsx (hunk 7 above)
Comment (FBalint) [reply src/platforms/Token/hooks/useCollectToken.tsx:62]: This one is specifically for the claim endpoints errors. Does `useFetcherWithSign` have an `onError` function as well?

Comment (BrickheadJohnny) [reply src/platforms/Token/hooks/useCollectToken.tsx]: Oh, my bad, then it's fine this way!

File: src/platforms/Token/EditTokenModal.tsx (hunk 6 above)
Comment (BrickheadJohnny) [reply src/platforms/Token/EditTokenModal.tsx:68]: Oh, I forgot about that. You're right, then let's leave these events here! 👌

**Pull request merged**